nautobot/extras/models/groups.py

```diff
@@ -354,6 +355,9 @@ def set_filter(self, form_data):
         # Use the auto-generated filterset form perform creation of the filter dictionary.
         filterset_form = filterset.form
 
+        # Get the declared form for any overloaded form field definitions.
+        declared_form = get_form_for_model(filterset._meta.model, form_prefix="Filter")
```

Comment:
Do we need to fallback to `filterset_form` if `get_form_for_model` fails?

Reply:
We do when we lookup the field below.